general_tools/log.py

```diff
@@ -50,3 +50,12 @@ def get_logger(name='NTHU OJ'):
     logger.addHandler(ch)
 
     return logger
+
+
+def get_client_ip(request):
```

Comment:
Indeed, ``utils`` is more succinct. I'll refactor about it.

It's clear that ``has_X_ownership`` should be in ``user_info.py``, and ``get_client_ip`` should be in ``network_info``. I'm not sure what category should ``get_logger`` be fallen into, though.